src/test/java/seedu/address/model/person/UniqueLogListTest.java

```diff
@@ -0,0 +1,148 @@
+        UniqueLogList uniqueLogList = new UniqueLogList(); // initially empty, test add separately
+        assertFalse(uniqueLogList.contains(toFind));
+    }
+
+    @Test
+    public void contains_logInList_returnsTrue() {
+        Log toAdd = new Log("title", null);
+        Log identical = new Log("title", "some description");
+        UniqueLogList uniqueLogList = new UniqueLogList();
+        uniqueLogList.addLog(toAdd);
+        assertTrue(uniqueLogList.contains(identical));
+    }
+
+    @Test
+    public void containsExactly_null_throwsNullPointerException() {
+        UniqueLogList uniqueLogList = new UniqueLogList();
+        assertThrows(NullPointerException.class, () -> uniqueLogList.containsExactly(null));
+    }
+
+    @Test
+    public void containsExactly_logNotInList_returnsFalse() {
+        Log toFind = new Log("title", null);
+        UniqueLogList uniqueLogList = new UniqueLogList(); // initially empty, test add separately
+        assertFalse(uniqueLogList.containsExactly(toFind));
+    }
+
+    @Test
+    public void containsExactly_logInList_returnsTrue() {
+        Log toAdd = new Log("title", "some description");
+        Log identical = new Log("title", "some description");
+        UniqueLogList uniqueLogList = new UniqueLogList();
+        uniqueLogList.addLog(toAdd);
+        assertTrue(uniqueLogList.containsExactly(identical));
+    }
+
+    @Test
+    public void add_nullLog_throwsNullPointerException() {
+        UniqueLogList uniqueLogList = new UniqueLogList();
+        assertThrows(NullPointerException.class, () -> uniqueLogList.addLog(null));
+    }
+
+
+    @Test
+    public void add_duplicateLog_throwsDuplicateLogException() {
+        Log toAdd = new Log("title", null);
+        Log similarLog = new Log("title", "some description"); // similar == same title
+        Log anotherSimilarLog = new Log("title", "some other description");
+        UniqueLogList uniqueLogList = new UniqueLogList();
+        uniqueLogList.addLog(toAdd);
+        assertThrows(DuplicateLogException.class, () -> uniqueLogList.addLog(similarLog));
+        assertThrows(DuplicateLogException.class, () -> uniqueLogList.addLog(anotherSimilarLog));
+    }
+
+    @Test
+    public void setLog_targetLogNotInside_throwsLogNotFoundException() {
+        Log target = new Log("target", null);
+        Log editedLog = new Log("target", "now added description");
+        Log notTargetLog = new Log("not target", null);
+
+        // target not in empty list
+        UniqueLogList uniqueLogList = new UniqueLogList(); // initially empty
+        assertThrows(LogNotFoundException.class, () -> uniqueLogList.setLog(target, editedLog));
+
+        // target not in list
+        uniqueLogList.addLog(notTargetLog); // contains non-target log
+        assertThrows(LogNotFoundException.class, () -> uniqueLogList.setLog(target, editedLog));
+    }
+
+
+    @Test
+    public void setLog_validEdit_contains_returnsTrue() {
```

Comment:
I don't think this adheres to the test method naming convention. Additionally, might be clearer to split this into two methods?